pyanaconda/modules/network/nm_client.py

```diff
@@ -0,0 +1,667 @@
+        method4 = "manual"
+    else:
+        method4 = "auto"
+
+    connection.remove_setting(NM.SettingIP4Config)
+    s_ip4 = NM.SettingIP4Config.new()
+    s_ip4.set_property(NM.SETTING_IP_CONFIG_METHOD, method4)
+    if method4 == "manual":
+        prefix4 = netmask2prefix(network_data.netmask)
+        addr4 = NM.IPAddress.new(socket.AF_INET, network_data.ip, prefix4)
+        s_ip4.add_address(addr4)
+        if network_data.gateway:
+            s_ip4.props.gateway = network_data.gateway
+    connection.add_setting(s_ip4)
+
+    # ipv6 settings
+    if network_data.noipv6:
+        method6 = "ignore"
+    elif not network_data.ipv6 or network_data.ipv6 == "auto":
+        method6 = "auto"
+    elif network_data.ipv6 == "dhcp":
+        method6 = "dhcp"
+    else:
+        method6 = "manual"
+
+    connection.remove_setting(NM.SettingIP6Config)
+    s_ip6 = NM.SettingIP6Config.new()
+    s_ip6.set_property(NM.SETTING_IP_CONFIG_METHOD, method6)
+    if method6 == "manual":
+        addr6, _slash, prefix6 = network_data.ipv6.partition("/")
+        if prefix6:
+            prefix6 = int(prefix6)
+        else:
+            prefix6 = 64
+        addr6 = NM.IPAddress.new(socket.AF_INET6, addr6, prefix6)
+        s_ip6.add_address(addr6)
+        if network_data.ipv6gateway:
+            s_ip6.props.gateway = network_data.ipv6gateway
+    connection.add_setting(s_ip6)
+
+    # nameservers
+    if network_data.nameserver:
+        for ns in [str.strip(i) for i in network_data.nameserver.split(",")]:
+            if NM.utils_ipaddr_valid(socket.AF_INET6, ns):
+                s_ip6.add_dns(ns)
+            elif NM.utils_ipaddr_valid(socket.AF_INET, ns):
+                s_ip4.add_dns(ns)
+            else:
+                log.error("IP address %s is not valid", ns)
+
+
+def bind_connection(nm_client, connection, bindto, device_name=None, bind_exclusively=True):
+    """Bind the connection to device name or mac address.
+
+    :param connection: uuid of the connection
+    :type connection: str
+    :param bindto: type of binding of the connection (mac address of device name)
+                    - BIND_TO_MAC for mac address
+                    - None for device name (default)
+    :type bindto: pykickstart --bindto constant
+    :param device_name: device name for binding
+    :type device_name: str
+    :param bind_exclusively: when binding to an entity, remove reference to the other
+    :type bind_exclusively: bool
+    """
+    msg = "bind connection {} to {}:".format(connection.get_uuid(), bindto or "iface")
+
+    s_con = connection.get_setting_connection()
+    if not s_con:
+        log.warning("%s no connection settings", msg)
+        return False
+    interface_name = s_con.get_interface_name()
+    s_wired = connection.get_setting_wired()
+    if s_wired:
+        mac_address = s_wired.get_mac_address()
+    else:
+        mac_address = None
+
+    # bind to mac address
+    if bindto == BIND_TO_MAC:
```

Comment:
Could you please abstract these big `if-else` parts.

Reply:
https://github.com/rhinstaller/anaconda/pull/1771/commits/f12340e370a347255ac04cbbbb080cc6f3f3d109